react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/views/ComposeViewWrapper.kt

```diff
@@ -2,14 +2,42 @@ package com.revenuecat.purchases.react.ui.views
 
 import android.content.Context
 import android.util.AttributeSet
+import android.util.Log
 import android.view.View
 import android.view.ViewGroup
 import android.widget.FrameLayout
+import androidx.lifecycle.ProcessLifecycleOwner
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import androidx.lifecycle.findViewTreeViewModelStoreOwner
+import androidx.lifecycle.setViewTreeLifecycleOwner
+import androidx.lifecycle.setViewTreeViewModelStoreOwner
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.findViewTreeSavedStateRegistryOwner
+import androidx.savedstate.setViewTreeSavedStateRegistryOwner
 
+/**
+ * Base wrapper for Compose views in React Native.
+ *
+ * Fixes the issue where Compose views crash in React Native modals with:
+ * - "ViewTreeLifecycleOwner not found"
+ * - "ViewTreeSavedStateRegistryOwner not found"
+ * - "No ViewModelStoreOwner was provided"
+ *
+ * React Native modals create isolated view hierarchies without the owners
+ * that Compose requires. This wrapper automatically sets all required owners
+ * on the modal containers.
+ *
+ * This wrapper also fixes a Cannot locate windowRecomposer error
+ * See: https://github.com/RevenueCat/react-native-purchases/issues/1162
+ */
 abstract class ComposeViewWrapper<T : View> : FrameLayout {
     protected var wrappedView: T? = null
     protected var isAttached = false
 
+    private val savedStateRegistryOwner: SavedStateRegistryOwner? by lazy { findSavedStateRegistryOwner() }
+    private val viewModelStoreOwner: ViewModelStoreOwner? by lazy { findViewModelStoreOwner() }
```

Comment:
I'm wondering, does it make sense to not store these and just calculate them on the fly in the `setComposeOwnersOnFrameLayoutContainerIfNeeded`? I guess it depends on whether we need to set it on the entire parent hierarchy, or only the immediate parent. If it's the first, this is good. If second, I would vote to just calculate it there.